frontend/lib/ui/artefact_dialog/artefact_build_expandable.dart

```diff
@@ -7,9 +7,14 @@ import '../spacing.dart';
 import 'test_execution_expandable.dart';
 
 class ArtefactBuildExpandable extends StatelessWidget {
-  const ArtefactBuildExpandable({super.key, required this.artefactBuild});
+  const ArtefactBuildExpandable({
+    super.key,
+    required this.artefactBuild,
+    required this.artefactId,
+  });
```

Comment:
I agree passing the prop is not perfect, but I wasn't sure how to do it otherwise.

We don't need the artefactId to send this request, only the test execution id, but we need the artefactID to get the ArtefactBuild object from riverpod.

I guess the best approach would be to have TestExecutionProvier which would not send requests to the backend but rather take the data from the ArtefactBuildsProvider, similarly to how we have the StageArtefactsProvider take the data from FamilyArtefactsProvider.

However, my only concern is how we would update the ArtefactBuildProvider state in that case, as it might include some incosistent data with the TestExecutionProvider, in case we change something. I guess in that case we need to make sure to never use the .testExecution object from ArtefactBuildProvider, but to always use the TestExecutionProvider, which makes sense.

I'm not sure about the flutter data package, as I don't have a strong understanding of how Flutter works yet. I will try to read more around it.

How do you suggest we go about this?